src/containers/questions-list/QuestionsList.tsx

```diff
@@ -0,0 +1,83 @@
+import { FC } from 'react'
+import {
+  DragDropContext,
+  Droppable,
+  Draggable,
+  DraggableProvided,
+  DraggableStateSnapshot,
+  DropResult,
+  DroppableProvided
+} from 'react-beautiful-dnd'
+import Box from '@mui/material/Box'
+
+import Question from '~/components/question/Question'
+
+import { styles } from '~/containers/questions-list/QuestionsList.styles'
+import { QuestionWithCategory } from '~/types'
+
+interface QuestionsListProps {
+  items: QuestionWithCategory[]
+  setItems: (items: QuestionWithCategory[]) => void
+}
+
+const QuestionsList: FC<QuestionsListProps> = ({ items, setItems }) => {
+  const reorder = (
+    list: QuestionWithCategory[],
+    startIndex: number,
+    endIndex: number
+  ): QuestionWithCategory[] => {
+    const result = Array.from(list)
+    const [removed] = result.splice(startIndex, 1)
+    result.splice(endIndex, 0, removed)
```

Comment:
can you return it immediately here?

Reply:
no, i can't cause `.splice` method modifies original array